R/oc_forward.R

```diff
@@ -0,0 +1,328 @@
+#'   Required. By default, \code{\link{oc_key}} will attempt to retrieve the key
+#'   from the environment variable \code{OPENCAGE_KEY}.
+#' @param bounds A list of bounding boxes, i.e. numeric vectors, each with 4
+#'   coordinates forming the south-west and north-east corners of a bounding box
+#'   \code{list(c(xmin, ymin, xmax, ymax))}. The \code{bounds} parameter will
+#'   restrict the possible results to the supplied region. It can easily be
+#'   specified with the \code{\link{oc_bbox}} helper, for example like
+#'   \code{bounds = oc_bbox(-0.563160, 51.280430, 0.278970, 51.683979)}.
+#' @param countrycode A two letter code as defined by the
+#'   \href{https://www.iso.org/obp/ui/#search/code}{ISO 3166-1 Alpha 2} standard
+#'   that restricts the results to the given country or countries. E.g. "AR" for
+#'   Argentina, "FR" for France, "NZ" for the New Zealand. Multiple countrycodes
+#'   per \code{placename} must be wrapped in a list.
+#' @param language An
+#'   \href{https://en.wikipedia.org/wiki/IETF_language_tag}{IETF language tag}
+#'   (such as "es" for Spanish or "pt-BR" for Brazilian Portuguese). OpenCage
+#'   will attempt to return results in that language. If it is not specified,
+#'   "en" (English) will be assumed by the API.
+#' @param limit The maximum number of results that should be returned. Integer
+#'   values between 1 and 100 are allowed, the default is 10 (\code{oc_forward})
+#'   or 1 (\code{oc_forward_df}), respectively.
+#' @param min_confidence An integer value between 0 and 10 indicating the
+#'   precision of the returned result as defined by it's geographical extent,
+#'   (i.e. by the extent of the result's bounding box). See the
+#'   \href{https://opencagedata.com/api#confidence}{API documentation} for
+#'   details. Only results with at least the requested confidence will be
+#'   returned.
+#' @param no_annotations A logical vector indicating whether additional
+#'   information about the result location should be returned. \code{TRUE} by
+#'   default, which means that the output will not contain annotations.
+#' @param no_dedupe A logical vector (default \code{FALSE}), when \code{TRUE}
+#'   the output will not be deduplicated.
+#' @param no_record A logical vector (default \code{FALSE}), when \code{TRUE} no
+#'   log entry of the query is created and the geocoding request is not cached
+#'   by OpenCage.
+#' @param abbrv A logical vector (default \code{FALSE}), when \code{TRUE}
+#'   addresses in the \code{formatted} field of the results are abbreviated
+#'   (e.g. "Main St." instead of "Main Street").
+#' @param add_request A logical vector (default \code{FALSE}) indicating whether
+#'   the request is returned again with the results. If the \code{return} value
+#'   is a \code{df_list}, the query text is added as a column to the results.
+#'   \code{json_list} results will contain all request parameters, including the
+#'   API key used! For \code{geojson_list} this is currently ignored by
```

Comment:
I wonder whether we should add even more warnings?

Reply:
Anything in particular, you have in mind. Throwing a warning/message seems a bit much? Or we could obfuscate the key, like e.g. vcr does?

Reply:
And users can suppress warnings if they want

Reply:
Maybe even `stopifnot(!interactive())` (or equivalent) if `output = "json_list"` and `add_request = TRUE`?

Reply:
👌